cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -1006,6 +1008,32 @@ public void grantKeyVaultAccessPolicyToServicePrincipal(String resourceGroupName
         });
     }
 
+    public boolean checkKeyVaultAccessPolicyForServicePrincipal(String resourceGroupName, String vaultName, String principalObjectId) {
+        return handleAuthException(() -> {
+            List<AccessPolicy> accessPolicies = azure.vaults()
+                    .getByResourceGroup(resourceGroupName, vaultName)
+                    .accessPolicies();
+            if (accessPolicies != null) {
+                for (int i = accessPolicies.size() - 1; i >= 0; i--) {
+                    AccessPolicyEntry accessPolicyEntry = accessPolicies.get(i).inner();
+                    if (checkKeyVaultAccessPolicyListForServicePrincipal(accessPolicyEntry, principalObjectId)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        });
+    }
+
+    @VisibleForTesting
+    public boolean checkKeyVaultAccessPolicyListForServicePrincipal(AccessPolicyEntry accessPolicyEntry, String principalObjectId) {
```

Comment:
No, it should be at maximum package-private as I mentioned last time. It makes no sense to have it annotated with `@VisibleForTesting` when it is `public`, but we do not really intend to make this `public`.

On the other hand, if the aforementioned code extraction in `checkKeyVaultAccessPolicyForServicePrincipal()` is done, it will not be worth testing `checkKeyVaultAccessPolicyListForServicePrincipal()` separately, so this method can be made `private`.